pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/RebalancePreChecker.java

```diff
@@ -22,10 +22,33 @@
 import java.util.concurrent.ExecutorService;
 import javax.annotation.Nullable;
 import org.apache.pinot.controller.helix.core.PinotHelixResourceManager;
+import org.apache.pinot.controller.util.TableSizeReader;
 import org.apache.pinot.spi.config.table.TableConfig;
 
 
 public interface RebalancePreChecker {
-  void init(PinotHelixResourceManager pinotHelixResourceManager, @Nullable ExecutorService executorService);
-  Map<String, String> check(String rebalanceJobId, String tableNameWithType, TableConfig tableConfig);
+  void init(PinotHelixResourceManager pinotHelixResourceManager, @Nullable ExecutorService executorService,
+      double diskUtilizationThreshold);
+
+  class TableFacts {
```

Comment:
nit: can we rename this to `PreCheckContext` since this is specifically used for pre-checks only.